src/kvazaar.h

```diff
@@ -34,6 +34,10 @@
 extern "C" {
 #endif
 
+#if defined(KVZ_STATIC_LIB) && defined(PIC)
+  #undef KVZ_STATIC_LIB
```

Comment:
I think we can use `defined(PIC)` to differentiate between shared and static builds as long as we check `defined(KVZ_DLL_EXPORTS)` to make sure that `PIC` is meant for kvazaar. Maybe something like this:
```c
#if defined(KVZ_DLL_EXPORTS)
  #if !defined(PIC)
    // Building static kvazaar library.
    #define KVZ_PUBLIC
  #elif defined(_WIN32) || defined(__CYGWIN__)
    // Building kvazaar DLL on Windows.
    #define KVZ_PUBLIC __declspec(dllexport)
  #elif defined(__GNUC__)
    // Building kvazaar shared library with GCC.
    #define KVZ_PUBLIC __attribute__ ((visibility ("default")))
  #else
    #define KVZ_PUBLIC
  #endif
#else
  #if defined(KVZ_STATIC_LIB)
    // Using static kvazaar library.
    #define KVZ_PUBLIC
  #elif defined(_WIN32) || defined(__CYGWIN__)
    // Using kvazaar DLL on Windows.
    #define KVZ_PUBLIC __declspec(dllimport)
  #else
    // Using kvazaar shared library and not on Windows.
    #define KVZ_PUBLIC
  #endif
#endif
```